### DISCUSSION TITLE
where can i find the tutorial for implementing python interface of dp in c

### DISCUSSION THREAD

**Discussion by kleinZhf** [deepmodeling/deepmd-kit#778] [Q&A]
I may want to use the dp model in c/cpp or fortran, but i find it quite confused to add the cpp interface of tensorflow. Instead i remember there used to be a simple python interface for c implement in dp tutorials, but i cannot find the pages now, could you pls offer a link for this?

**Comment by Ericwang6** [comment]
Sorry, there might be lack of info about C interface in the document of 1.x version, but there is one in 2.0 version, https://github.com/deepmodeling/deepmd-kit/blob/devel/doc/getting-started.md#c-interface. Please notice that the interface of 1.x and 2.0 are quite different.

**Reply by kleinZhf** [reply]
this cpp interface seemed to require tensorflow cpp interface for compiling, do you have the latest tutorial for install the tensorflow cpp interface? And what if i need to run my code on a server which does not have direct access to public network, is there any advice for install the tensorflow cpp interface offline?

**Reply by Ericwang6** [reply]
Compiling C++ interface for tensorflow is quite complicated.....There is one solution that you can install deepmd-kit via off-line packages so that C++ interface of tensorflow will be automatically installed.

**Reply by kleinZhf** [reply]
any link guide to this offline packages?

**Reply by Ericwang6** [reply]
https://github.com/deepmodeling/deepmd-kit/releases
Just download the package and run `bash xxx.sh` to install, and follow the instructions during your installation. The program will ask you to enter a directory to set up a conda environment, and the C++ interface will be installed under the conda environment directory
